packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -3177,7 +3177,12 @@ class _MenuLayout extends SingleChildLayoutDelegate {
       } else if (offBottom(y)) {
         final double newY = anchorRect.top - childSize.height;
         if (!offTop(newY)) {
-          y = newY;
+          // Only move the menu up if it is the root menu.
```

Comment:
This comment doesn't appear to be related to the conditional, since there can be menus that are horizontal that aren't root menus.  Sure, it's weird, but it's possible (you could have a menu bar that appears when you open another menu item).

Probably you just want to adjust the comment to talk about horizontal menus, but if you did want to actually only do it if it's the root menu, you would have to pass that information in to this layout widget.

Reply:
Ok, updated the comment. I tested the behaviour of having a MenuBar as a child of a SubmenuButton in a MenuAnchor widget and it's behaving as it should.